src/Temperature-Accretech_UFseries/main.py

```diff
@@ -113,26 +119,46 @@ def connect(self) -> None:
         #     self.device_communication[self.instance_key] = self.port
 
         self.prober = accretech_uf.AccretechProber(self.port)
-
         self.prober.set_verbose(self.verbosemode)
 
+        self.unique_identifier = "Driver_Accretech_UFseries_" + self.port_str
+        if not self.unique_identifier in self.device_communication:
+            self.device_communication[self.unique_identifier] = None
+
     def disconnect(self) -> None:
         """Disconnect from the device."""
-        del self.prober  # this makes sure the event mechanism is disabled
+        if self.unique_identifier in self.device_communication:
+            self.prober.unregister_srq_event()  # this makes sure the event mechanism is disabled
+            del self.device_communication[self.unique_identifier]
+        del self.prober  
 
     def initialize(self) -> None:
         """Initialize the device."""
-        # TODO: Check
         self.prober.reset_alarm()
 
+    def deinitialize(self):
+        pass
+
+    def unconfigure(self):
+        if self.idle_temperature != "":
+            _, target_temperature = self.prober.request_chuck_temperature()
+            if target_temperature != float(self.idle_temperature):
+                self.prober.set_chuck_temperature(float(self.idle_temperature))
+            else:
+                debug("Target temperature already set.")
+        
     def apply(self) -> None:
         """Apply the given temperature."""
-        self.prober.set_chuck_temperature(self.value)
+        _, target_temperature = self.prober.request_chuck_temperature()
+        if target_temperature != float(self.value):
+            self.prober.set_chuck_temperature(self.value)
+        else:
+            debug("Target temperature already set.")
```

Comment:
Might as well be a debugmode only print?

Reply:
see above